api/api.go

```diff
@@ -0,0 +1,127 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/fluidkeys/api/v1structs"
+)
+
+const (
+	defaultBaseURL = "https://api.fluidkeys.com/v1/"
+	userAgent      = "fluidkeys"
+)
+
+// A Client manages communication with the Fluidkeys Server API.
+type Client struct {
+	client    *http.Client // HTTP client used to communicate with the API.
+	BaseURL   *url.URL     // Base URL for API requests
+	UserAgent string       // User agent used when communicating with the  API.
+}
+
+// NewClient returns a new Fluidkeys Server API client. If a nil httpClient is
+// provided, http.DefaultClient will be used.
+func NewClient() *Client {
+	baseURL, _ := url.Parse(defaultBaseURL)
+
+	return &Client{
+		client:    http.DefaultClient,
+		BaseURL:   baseURL,
+		UserAgent: userAgent,
+	}
+}
+
+// GetPublicKey attempts to get a single armorded public key.
+func (c *Client) GetPublicKey(email string) (string, *http.Response, error) {
+	url := fmt.Sprintf("email/%s/key", url.QueryEscape(email))
```

Comment:
Being picky, it's not a `url` (`https://example.com/v1/email`) it's *part* of the path of a full URL. Suggest using `path` instead of `url`